src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -930,7 +930,7 @@ public override object VisitField(FieldSymbol symbol, TypeCompilationState argum
                     var lazySemanticModel = body == null ? null : new Lazy<SemanticModel>(() =>
                     {
                         var syntax = body.Syntax;
-                        var semanticModel = (CSharpSemanticModel)_compilation.GetSemanticModel(syntax.SyntaxTree);
+                        var semanticModel = (CSharpSemanticModel)AnalyzerDriver.GetOrCreateCachedSemanticModel(syntax.SyntaxTree, _compilation, _cancellationToken);
```

Comment:
_The creation of a fresh semantic model was intentional_
This works fine from the perspective of the command line builds. However, with the new compilation events based IDE analyzer driver, the IDE already has a semantic model which needs to be used across diagnostic computations for different analyzers. This is the semantic model on which CompilationWithAnalyzers explicitly invokes GetDiagnostics to force binding symbols in the document, and passes down to the analyzers.

I think the right fix is to change [this](http://source.roslyn.io/Microsoft.CodeAnalysis.CSharp/R/98476126a76c7324.html) API to take a semantic model instead of the tree (all callsites are from with semantic model, so should be a trivial change), and thread that semantic model down to the MethodBodyCompiler invocation [here](http://source.roslyn.io/Microsoft.CodeAnalysis.CSharp/Compiler/MethodCompiler.cs.html#ff4d26babc5a460e).
The code being changed here can then just check if a non-null semantic model was passed in, and if so use that one (IDE case), otherwise create a fresh semantic model (command line case)